go/types/graph_builder.go

```diff
@@ -112,19 +113,23 @@ type graphOpContainer struct {
 // object.
 func (b *GraphBuilder) Build() Value {
 	var opc opCache
+	var opcStore opCacheStore
 
-	checkFirstCall := func() {
+	defer func() {
+		opcStore.destroy()
+	}()
+
+	// Use function here to take advantage fo the deferred call to mutex.Unlock()
+	func() {
 		b.mutex.Lock()
```

Comment:
i'm confused about the threadsafety of this object. This mutex doesn't appear to be used anywhere else. Is the assumption that this will only be used from a single goroutine? Is that a *true* assumption? 

Seems weird to lock here and nowhere else?